packages/grid/data-grid/src/DataGrid.test.tsx

```diff
@@ -4,28 +4,37 @@ import PropTypes from 'prop-types';
 import { createClientRender, ErrorBoundary } from 'test/utils';
 import { useFakeTimers } from 'sinon';
 import { expect } from 'chai';
-import { DataGrid } from '@material-ui/data-grid';
+import { DataGrid, GridState } from '@material-ui/data-grid';
+import { getColumnValues } from 'test/utils/helperFn';
 
 describe('<DataGrid />', () => {
   const render = createClientRender();
+
   const defaultProps = {
     rows: [
       {
         id: 0,
         brand: 'Nike',
       },
+      {
+        id: 1,
+        brand: 'Adidas',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+      },
     ],
     columns: [{ field: 'brand', width: 100 }],
   };
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
```

Comment:
I'm not a fan of skipping all the tests on jsdom. There are some parts of the logic that can work without the layout, like the warnings, the composition API, and probably more going forward. I was also hoping that with a few mocking, we could simulate a full layout on jsdom. The main advantage of jsdom is the speed, you don't have to build the whole test. I believe it runs faster, especially during watch mode.

Reply:
it's the same setup as XGrid.

Reply:
I don't see the link with XGrid.

Reply:
If we do this, the set timeout added for the warning in the resize handle becomes dead code and should be removed.

*master*

<img width="176" alt="Capture d’écran 2020-11-10 à 11 03 24" src="https://user-images.githubusercontent.com/3165635/98659436-638f0e00-2344-11eb-9210-791d174636af.png">

*PR*

<img width="176" alt="Capture d’écran 2020-11-10 à 11 03 48" src="https://user-images.githubusercontent.com/3165635/98659490-7275c080-2344-11eb-9e28-74d6aa8b54cf.png">

Reply:
https://github.com/mui-org/material-ui-x/blob/master/packages/grid/x-grid/src/XGrid.test.tsx#L69

Reply:
Again, XGrid has all its tests that require the layout, its not the case of DataGrid.